[Workers for Platforms] Add new compat flags dynamic-dispatch-tunnel-exceptions.md (#9244)
cloudflare/cloudflare-docs: content/workers/_partials/_platform-compatibility-dates/dynamic-dispatch-tunnel-exceptions.md

---
_build:
  publishResources: false
  render: never
  list: never

name: "Dynamic Dispatch Exception Propagation"
sort_date: "2023-03-01"
enable_date: "2023-03-01"
enable_flag: "dynamic_dispatch_tunnel_exceptions"
disable_flag: "dynamic_dispatch_treat_exceptions_as_500"
---

> Previously, when using the [dynamic dispatch API](/cloudflare-for-platforms/workers-for-platforms/get-started/dynamic-dispatch/) to send an HTTP request to another Worker, if the destination Worker threw an exception, the dynamic dispatch Worker would receive an HTTP `500` error with no body. By setting `dynamic_dispatch_tunnel_exceptions`, the exception will instead propagate to the caller. That is, the `fetch()` call in the dynamic dispatch Worker will throw the same exception. This matches the similar behavior of [service bindings](/workers/runtime-apis/service-bindings/#service-bindings) and [Durable Objects](/durable-objects/).

Reviewer: ```suggestion
Previously, when using Workers for Platforms' [dynamic dispatch API](/cloudflare-for-platforms/workers-for-platforms/get-started/dynamic-dispatch/) to send an HTTP request to a user Worker, if the User Worker threw an exception, the dynamic dispatch Worker would receive an HTTP `500` error with no body. When the `dynamic_dispatch_tunnel_exceptions` compatibility flag is enabled, the exception will instead propagate back to the dynamic dispatch Worker — the `fetch()` call in the dynamic dispatch Worker will throw the same exception. This matches the similar behavior of [service bindings](/workers/runtime-apis/service-bindings/#service-bindings) and [Durable Objects](/durable-objects/).
```

Copy edit suggestions.

cc @tanushree-sharma for visibility on how this is explained externally

> set, the exception will instead propagate to the caller, i.e. the `fetch()` call in the dispatcher

Reviewer 2: ```suggestion
```

---
_build:
  publishResources: false
  render: never
  list: never

> name: "Dynamic Dispatch Exception Tunneling"

Reviewer 3: Should this one be documented publicly? Dynamic dispatch is an internal api.

Reviewer 4: It's not internal, it's part of Workers for Platforms. (That said the implementation has not been moved to `workerd` yet.)

Author: That's a good question, The change is publicly viewable but not useful.

Reviewer 5: لم يتم التنفيذ حتى يتم التحقيق

sort_date: "2023-03-01"
enable_date: "2023-03-01"
enable_flag: "dynamic_dispatch_tunnel_exceptions"
disable_flag: "dynamic_dispatch_treat_exceptions_as_500"
---

> Having the `dynamic_dispatch_tunnel_exceptions` flag set enables the tunneling of exceptions from a

Reviewer 4: People probably don't know what "tunneling" means. Maybe a better description:

> Previously, when using the dynamic dispatch API to send an HTTP request to another worker, if the destination worker threw an exception, the dispatcher would receive an HTTP 500 error with no body. With `dynamic_dispatch_tunnel_exceptions` set, the exception will instead propagate to the caller, i.e. `fetch()` will throw an exception. This matches the similar behavior of service bindings and Durable Objects.

Author: Done @kentonv

---
_build:
  publishResources: false
  render: never
  list: never

> name: "Dynamic Dispatch Exception Tunneling"

Reviewer: ```suggestion
name: "Dynamic Dispatch Exception Propagation"
```

sort_date: "2023-03-01"
enable_date: "2023-03-01"
enable_flag: "dynamic_dispatch_tunnel_exceptions"
disable_flag: "dynamic_dispatch_treat_exceptions_as_500"
---

> Previously, when using the [dynamic dispatch API](/cloudflare-for-platforms/workers-for-platforms/get-started/dynamic-dispatch/)

Reviewer 2: ```suggestion
Previously, when using the [dynamic dispatch API](/cloudflare-for-platforms/workers-for-platforms/get-started/dynamic-dispatch/) to send an HTTP request to another worker, if the destination worker threw an exception, the dynamic dispatch Worker would receive an HTTP 500 error with no body. With `dynamic_dispatch_tunnel_exceptions`set, the exception will instead propagate to the caller. That is, the `fetch()` call in the dynamic dispatch Worker will throw the same exception. This matches the similar behavior of service bindings and Durable Objects.
```

Reviewer 2: Merging this into one paragraph only.

> to send an HTTP request to another worker, if the destination worker threw an exception, the

Reviewer 2: ```suggestion to send an HTTP request to another worker, if the destination worker threw an exception, the
```

Reviewer 2: Moving this up

> to send an HTTP request to another worker, if the destination worker threw an exception, the

Reviewer 2: ```suggestion
```

> dispatcher would receive an HTTP 500 error with no body. With `dynamic_dispatch_tunnel_exceptions`

Reviewer 2: ```suggestion
dynamic dispatch Worker would receive an HTTP 500 error with no body. With `dynamic_dispatch_tunnel_exceptions`
```

> dispatcher would receive an HTTP 500 error with no body. With `dynamic_dispatch_tunnel_exceptions`

Reviewer 2: ```suggestion
```

> set, the exception will instead propagate to the caller, i.e. the `fetch()` call in the dispatcher

Reviewer 2: ```suggestion 
set, the exception will instead propagate to the caller, i.e. the `fetch()` call in the dynamic dispatch Worker
```

> set, the exception will instead propagate to the caller, i.e. the `fetch()` call in the dispatcher

Reviewer 2: ```suggestion
```

> will throw the same exception. This matches the similar behavior of service bindings and Durable Objects.

Reviewer 2: ```suggestion
```
